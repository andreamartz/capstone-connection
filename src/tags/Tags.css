.tag {
  margin-right: .65rem;
}
.HTML {
  color: #9c27b0;
}
.CSS {
  color: #ff4025;
}
.JS {
  color: #ff9c2a;
}
.API {
  color: #006400;
}