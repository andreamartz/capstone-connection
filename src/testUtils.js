import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  bio: "This is my bio",
  email: "test@test.net",
  photoUrl: null,
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
