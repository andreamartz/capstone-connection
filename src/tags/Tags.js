import React from "react";
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import "./Tags.css";


const Tags = ({ tags }) => {
  return (
    <Box display="flex">
      {tags
        ? (
          <Box display="flex">
            {tags.map(tag => (
              <span
                key={tag.id}
                className="tag"
              >
                <Typography className={tag.text}>
                  {tag.text}
                </Typography>
              </span>
            ))}
          </Box>
        ) : (
          null
        )
      }
    </Box>
  )
};

export default Tags;