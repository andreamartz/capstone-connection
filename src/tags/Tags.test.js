import React from 'react';
import { render } from '@testing-library/react';
import Tags from './Tags';
import { UserProvider } from '../testUtils';

const tags = [
	{ id: 1, text: 'HTML' },
	{ id: 2, text: 'CSS' },
];

test('it renders without crashing', function () {
	render(
		<UserProvider>
			<Tags tags={tags} />
		</UserProvider>,
	);
});

test('it matches snapshot', function () {
	const { asFragment } = render(
		<UserProvider>
			<Tags tags={tags} />
		</UserProvider>,
	);
	expect(asFragment()).toMatchSnapshot();
});
