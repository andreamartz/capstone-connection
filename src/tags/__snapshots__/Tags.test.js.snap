// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <span
        class="tag"
      >
        <p
          class="MuiTypography-root HTML MuiTypography-body1"
        >
          HTML
        </p>
      </span>
      <span
        class="tag"
      >
        <p
          class="MuiTypography-root CSS MuiTypography-body1"
        >
          CSS
        </p>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
